libtcmu.c

```diff
@@ -491,8 +492,9 @@ static int add_device(struct tcmulib_context *ctx, char *dev_name,
 	}
 
 	snprintf(str_buf, sizeof(str_buf), "/dev/%s", dev_name);
+	dev->mmap_name = strndup(str_buf, sizeof(str_buf));
```

Comment:
Maybe we can just delete the str_buf here and use asprintf(&dev->map_name, "/dev/%s", dev_name);

Reply:
Also add a check for the failover case. You will probably want to rename the err_free tag to err_free_dev.

If (asprintf failed)
     goto err_free_dev;

And then add a new err_free_mmap_name goto and fix up the existing goto err_free lines.